### PR TITLE
improve summary

### LINKED ISSUE #125: fatal: refusing to merge unrelated histories
I'm getting the `fatal: refusing to merge unrelated histories` when I'm trying to merge branches, for some reason.

My steps:
```yml
- name: Use checkout v2
  uses: actions/checkout@v2
  with:
    ref: refs/heads/master
    token: ${{ secrets.GITHUB_CI_TOKEN }}
- name: Merge development into master
  run: |
    echo "# Config"
    git config --local user.email "[hidden]"
    git config --local user.name "CI"
    echo "# Fetch all"
    git fetch --all
    echo "# Checkout/pull development"
    git checkout development
    git pull
    echo "# Checkout/pull master"
    git checkout master
    git pull
    echo "# Merge"
    git merge development -X theirs
    echo "# Push"
    git push origin master
```

The error:
image

`master` and `development` branches can't have unrelated histories, because `development` was branched from `master`.

Running those commands locally works as expected.

**Update:**
Adding `--allow-unrelated-histories` to the merge command makes it work.
I don't understand how does git detect the branches as having different histories though.

### LINKED ISSUE #126: Not a git repository error with checkout v2
See [gitfool/Cake.Dungeon](https://github.com/gitfool/Cake.Dungeon) workflow builds for commit [6a287dc](https://github.com/gitfool/Cake.Dungeon/commit/6a287dce8d1c137ad4ca3c3cd566396036f0570b):
- [Linux](https://github.com/gitfool/Cake.Dungeon/runs/366247199)
- [Mac](https://github.com/gitfool/Cake.Dungeon/runs/366247194)
- [Windows](https://github.com/gitfool/Cake.Dungeon/runs/366247210)
- [Docker](https://github.com/gitfool/Cake.Dungeon/runs/366248444)

The Docker build is failing with error:
```
fatal: not a git repository (or any parent up to mount point /)
```
This [seems](https://github.com/gitfool/Cake.Dungeon/runs/366248444#step:3:23) to be because, after I switched from checkout v1 to checkout v2, the git version used regressed from [2.24.0](https://github.com/gitfool/Cake.Dungeon/runs/349851037#step:3:12) to [2.17.1](https://github.com/gitfool/Cake.Dungeon/runs/366248444#step:3:20). All other builds above are still using git version 2.24.0.

Did something change on your end or am I missing something? 🤔

### PR DESCRIPTION
**Pull request by ericsciple**
fixes #125 
fixes #126

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,29 +4,28 @@
...
-Refer [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn which commit `$GITHUB_SHA` points to for different events.
+The auth token is persisted in the local git config. This enables your scripts to run authenticated git commands. The token is removed during post-job cleanup. Use the input `persist-credentials` to opt-out.
+
+When Git 2.18 or higher is not in your PATH, falls back to the REST API to download the files.
 
 # What's new
 
-- Improved fetch performance
-  - The default behavior now fetches only the commit being checked-out
+- Improved performance
```
Comment (ericsciple) [inline README.md]: I simplified the `What's new` section to make it easier to skim

Also the summary is more verbose now, so I didn't want to repeat all details.

File: README.md [hunk 2]
```diff
@@ -6,27 +6,26 @@
 
 This action checks-out your repository under `$GITHUB_WORKSPACE`, so your workflow can access it.
```
Comment (ericsciple) [inline README.md]: The existing summary does not have enough details and folks are getting tripped up by the fetch-depth 1 default

File: README.md [hunk 3]
```diff
@@ -6,27 +6,26 @@
...
 
-By default, the repository that triggered the workflow is checked-out, for the ref/SHA that triggered the event.
+Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set `fetch-depth` to fetch more history. Refer [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn which commit `$GITHUB_SHA` points to for different events.
 
-Refer [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn which commit `$GITHUB_SHA` points to for different events.
+The auth token is persisted in the local git config. This enables your scripts to run authenticated git commands. The token is removed during post-job cleanup. Use the input `persist-credentials` to opt-out.
```
Comment (bryanmacfarlane) [inline README.md]: a nit but should it say set `persist-credentials` to false to opt-out?

File: README.md [hunk 4]
```diff
@@ -6,27 +6,26 @@
...
+  - Auth token persisted in the local git config
 - Creates a local branch
   - No longer detached HEAD when checking out a branch
-  - A local branch is created with the corresponding upstream branch set
 - Improved layout
   - The input `path` is always relative to $GITHUB_WORKSPACE
   - Aligns better with container actions, where $GITHUB_WORKSPACE gets mapped in
 - Fallback to REST API download
   - When Git 2.18 or higher is not in the PATH, the REST API will be used to download the files
+  - When using a job container, the container's PATH is used
```
Comment (ericsciple) [inline README.md]: this tripped up a customer migrating from v1

**Pull request merged**